lib/flappy_translator.dart

```diff
@@ -48,31 +51,52 @@ const List<String> RESERVED_WORDS = [
 ];
 
 class FlappyTranslator {
-  void generate(String filePath, {String targetPath = ""}) async {
-    final File file = File(filePath);
-
+  void generate(
+    String inputFilePath, {
+    String outputDir,
+    String fileName,
+    String className,
+    String delimiter,
+    int startIndex,
+    bool dependOnContext,
+    bool useSingleQuotes,
+  }) async {
+    final File file = File(inputFilePath);
     if (!file.existsSync()) {
-      FlappyLogger.logError("File $filePath does not exist");
+      FlappyLogger.logError("File $inputFilePath does not exist");
       return;
     }
 
-    String template = templateString;
+    // if null has been passed in, ensure that vars are given default values
+    outputDir ??= defaultOutputDirectory;
+    fileName ??= defaultFileName;
+    className ??= defaultClassName;
+    delimiter ??= defaultDelimiter;
+    startIndex ??= defaultStartIndex;
+    dependOnContext ??= defaultDependOnContext;
+    useSingleQuotes ??= defaultUseSingleQuotes;
+
+    // construct the template
+    var template =
+        templateBegining + (dependOnContext ? templateDependContext : templateDontDependContext) + templateEnding;
+    template = template.replaceAll(CLASS_NAME_TEMPLATE_KEY, className);
 
-    final FileParser fileParser = CSVParser();
+    final csvParser = CSVParser(fieldDelimiter: delimiter);
```

Comment:
I know that Dart does type inference, but I prefer to keep the types of the variables in order to make the code more readable and, so, maintainable

Reply:
Ok, updated all inferred types to be explicit.